Makefile: thanks MATSUI Fe2+ Tetsushi for catching, and for the
inspiration to do a dirty trick such that we don't inject a linker flag
on a linker that doesn't grok it (such as Darwin's odcctools)

--- Makefile
+++ Makefile
@@ -27,7 +27,7 @@
 CP        := cp
 
 ifndef STATIC
-LDFLAGS   += -Wl,--export-dynamic
+LDFLAGS   += $(ld --export-dynamic 2>&1 | fgrep -v unknown > /dev/null && echo -Wl,--export-dynamic)
 LIBADD    += $(shell echo | $(CC) -dM -E - | grep -q ' __linux__' && echo '-ldl')
 else
 CFLAGS    += -DSTATIC

main.c: Addition of prefix to various paths

--- main.c
+++ main.c
@@ -51,15 +51,15 @@
 char pretend = 0;
 char reinitialize = 0;
 char reinitialize_metacache = 0;
-char portdir[_Q_PATH_MAX] = "/usr/portage";
+char portdir[_Q_PATH_MAX] = "@GENTOO_PORTAGE_EPREFIX@/usr/portage";
 char portarch[20] = "";
 char portvdb[] = "var/db/pkg";
 char portcachedir[] = "metadata/cache";
-char portroot[_Q_PATH_MAX] = "/";
-char config_protect[_Q_PATH_MAX] = "/etc/";
+char portroot[_Q_PATH_MAX] = "@GENTOO_PORTAGE_EPREFIX@";
+char config_protect[_Q_PATH_MAX] = "@GENTOO_PORTAGE_EPREFIX@/etc/";
 
-char pkgdir[512] = "/usr/portage/packages/";
-char port_tmpdir[512] = "/var/tmp/portage/";
+char pkgdir[512] = "@GENTOO_PORTAGE_EPREFIX@/usr/portage/packages/";
+char port_tmpdir[512] = "@GENTOO_PORTAGE_EPREFIX@/var/tmp/portage/";
 
 char binhost[1024] = PORTAGE_BINHOST;
 char features[2048] = "noman noinfo nodoc";
@@ -432,7 +432,7 @@
 	char buf[BUFSIZE], *s, *p;
 
 	char profile[_Q_PATH_MAX], portage_file[_Q_PATH_MAX];
-	const char *files[] = {portage_file, "/etc/make.globals", "/etc/make.conf"};
+	const char *files[] = {portage_file, "@GENTOO_PORTAGE_EPREFIX@/etc/make.globals", "@GENTOO_PORTAGE_EPREFIX@/etc/make.conf"};
 	typedef enum { _Q_BOOL, _Q_STR, _Q_ISTR } var_types;
 	struct {
 		const char *name;
@@ -459,8 +459,8 @@
 			strncat(portroot, "/", sizeof(portroot));
 
 	f = 0;
-	if (readlink("/etc/make.profile", profile, sizeof(profile)) == -1)
-		strcpy(profile, "/etc/make.profile");
+	if (readlink("@GENTOO_PORTAGE_EPREFIX@/etc/make.profile", profile, sizeof(profile)) == -1)
+		strcpy(profile, "@GENTOO_PORTAGE_EPREFIX@/etc/make.profile");
 	if (profile[0] != '/') {
 		memmove(profile+5, profile, strlen(profile));
 		memcpy(profile, "/etc/", 5);
@@ -999,7 +999,7 @@
 
 #ifdef ENABLE_NLS	/* never tested */
 	setlocale(LC_ALL, "");
-	bindtextdomain(argv0, "/usr/share/locale");
+	bindtextdomain(argv0, "@GENTOO_PORTAGE_EPREFIX@/usr/share/locale");
 	textdomain(argv0);
 #endif
 
